### DISCUSSION TITLE
MongoDB error Server selection timeout: No available servers.

### DISCUSSION THREAD

**Discussion by inmancosta** [prisma/prisma#21754] [Q&A]
### Question

Did anyone experience this issue? I saw this previous [post](https://github.com/prisma/prisma/discussions/11929) , but my error seems to be slightly different:

### How to reproduce (optional)

4. See error:
<img width="929" alt="Screenshot 2023-11-02 at 1 21 37 PM" src="https://github.com/prisma/prisma/assets/120146242/7334aa69-3f53-41dc-b0c3-c033a3177860">

-->

### Expected behavior (optional)

I expected this data to be sent to the Mongo Db database, but it obviously didn't. Ive tried adding the directconnection=True into my mongo connection link along with double-checking my username/password. Still haven't fixed it.  Im also building this site based off this video: https://www.youtube.com/watch?v=K4ziF0MhbLc&t=3612s .

Ive reached 1:15 , if anyone thinks they can catch my error.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by janpio** [comment]
How did you react to the error message? It tells you that you are using an invalid connection string.

**Reply by inmancosta** [reply]
I double checked the password and username were correct in my .env file. Thats all I've done so far.  Here is my connection string

<code>DATABASE_URL="mongodb+srv://inmancosta1:password@cluster0.xeucpem.mongodb.net/ecommerce?retryWrites=true&w=majority"</code>

**Reply by inmancosta** [reply]
which file should i Be looking at given this invalid connection string?

**Reply by janpio** [reply]
It's interesting that the error message from the MongoDB driver talks about `directConnection=true` but your connection string does not actually include that. Weird.

**Reply by inmancosta** [reply]
Oh yea. I added that briefly, but it didn't work either. I got an error message with both .

**Reply by janpio** [reply]
But did you get the error message you shared when that was present or not? The screenshot in your original comment shows that in the error message.

**Reply by inmancosta** [reply]
Yes. Other than that indication, the error message was the same, indicating there weren't any mongoDb servers available...

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
